map/src/format.rs

```diff
@@ -239,31 +239,33 @@ pub struct MapItemLayerV1TilemapExtraRace {
     pub data: i32,
 }
 impl MapItemLayerV1TilemapExtraRace {
-    pub fn offset(version: i32, flags: u32) -> Option<usize> {
-        let offset = match version {
-            2 => MapItemLayerV1TilemapV2::sum_len(),
-            3 => MapItemLayerV1TilemapV3::sum_len(),
-            _ => return None,
+    pub fn offset(version: i32, flags: i32)
+        -> Result<usize, LayerTilemapError>
+    {
+        let offset = if version < 2 {
+            return Err(LayerTilemapError::InvalidVersion(version))
+        } else if version == 2 {
+            MapItemLayerV1TilemapV2::sum_len()
+        } else {
+            MapItemLayerV1TilemapV3::sum_len()
```

Comment:
I must say that I'm not a fan of lax parsing, trying to recover from bad values we don't know.

On the one hand, if someone actually increased the tilemap versions, introducing new fields, this could would break in inexplicable ways.

On the other hand, one could argue that no one could do this because it would break the DDNet parsing code. What do you think, would it be easier to change DDNet code?

Other possibilities include: Emit some kind of warning on tilemap version >= 2 (hard to do, we don't have a warning infrastructure in place for that); introduce some kind of lax parsing mode that just copies whatever DDNet is doing (also annoying as it's one more parameter to pass around).

Reply:
I could find only one map affected out of DDNet's (def-'s) maps7 collection. That's the map that is mentioned in the issue.

I also can't open it in DDNet's editor. I think we could declare this map corrupted and settle for an improved error reporting (I'll submit a separate pull request).

Reply:
It's also possible that it's valid for some modded server & editor - but I'm not sure if we want to try to find which mod that was and add support for it.